src/components/card-with-button/СardWithButton.js

```diff
@@ -0,0 +1,58 @@
+import { Box, Button } from '@mui/material'
+import { useContext, useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ModalContext } from '~/context/modal-context'
+import CardWithImage from '~/components/card-with-image/CardWithImage'
+import SignupDialog from '~/containers/guest-home-page/signup-dialog/SignupDialog'
+
+export const styles = {
+  wrap: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  }
+}
+
+const titleVariant = {
+  xs: 'h6'
+}
+
+const descriptionVariant = {
+  xs: 'subtitle2'
+}
+
+const CardWithButton = ({ array, role, btnText }) => {
+  const { t } = useTranslation()
+  const { setModal } = useContext(ModalContext)
+
+  const openDialog = (type) => {
+    setModal(<SignupDialog type={ type } />)
+  }
+
+  const cards = useMemo(

```

Comment:
The component has the name "Card with the button" it is unclear why we have something to map here and have cards.

Reply:
fixed